dev_tools/nyaml2nxdl/nyaml2nxdl_backward_tools.py

```diff
@@ -0,0 +1,1046 @@
+                        "{indent}{value}: \n".format(
+                            indent=(itm_depth) * DEPTH_SIZE, value=child.attrib["value"]
+                        )
+                    )
+
+                    if list(child):
+                        for item_doc in list(child):
+                            if remove_namespace_from_tag(item_doc.tag) == "doc":
+                                item_doc_depth = itm_depth + 1
+                                self.handle_not_root_level_doc(
+                                    item_doc_depth,
+                                    item_doc.text,
+                                    item_doc.tag,
+                                    file_out,
+                                )
+                            if (
+                                remove_namespace_from_tag(item_doc.tag) == CMNT_TAG
+                                and self.include_comment
+                            ):
+                                self.handel_comment(itm_depth + 1, item_doc, file_out)
+                if tag == CMNT_TAG and self.include_comment:
+                    self.handel_comment(itm_depth + 1, child, file_out)
+        else:
+            enum_list = ""
+            remove_nodes = []
+            for item_child in list(node):
+                tag = remove_namespace_from_tag(item_child.tag)
+                if tag == ("item"):
+                    enum_list = enum_list + "{value}, ".format(
+                        value=item_child.attrib["value"]
+                    )
+                if tag == CMNT_TAG and self.include_comment:
+                    self.handel_comment(depth, item_child, file_out)
+                    remove_nodes.append(item_child)
+            for ch_node in remove_nodes:
+                node.remove(ch_node)
+
+            file_out.write(
+                "{indent}{tag}: [{enum_list}]\n".format(
+                    indent=depth * DEPTH_SIZE,
+                    tag=remove_namespace_from_tag(node.tag),
+                    enum_list=enum_list[:-2] or "",
+                )
+            )
+
+    def handle_attributes(self, depth, node, file_out):
+        """Handle the attributes parsed from the xml file"""
+
+        allowed_attr = [
+            "name",
+            "type",
+            "units",
+            "nameType",
+            "recommended",
+            "optional",
+            "minOccurs",
+            "maxOccurs",
+            "deprecated",
+        ]
+
+        name = ""
+        node_attr = node.attrib
+        if "name" in node_attr:
+            pass
+        else:
+            raise ValueError("Attribute must have an name key.")
+        rm_key_list = []
+        # Maintain order: name and type in form name(type) or (type)name that come first
+        for key, val in node_attr.items():
+            if key == "name":
+                name = val
+                rm_key_list.append(key)
+
+        for key in rm_key_list:
+            del node_attr[key]
```

Comment:
`node_attr` acts like a [dictionary](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.attrib) so use `node_attr.get` and `remove`

Reply:
:+1: